tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -183,16 +187,24 @@ impl LeaderTpuCache {
         }
     }
 
-    pub fn extract_cluster_tpu_sockets(
+    fn extract_cluster_tpu_sockets(
+        protocol: Protocol,
         cluster_contact_info: Vec<RpcContactInfo>,
     ) -> HashMap<Pubkey, SocketAddr> {
         cluster_contact_info
             .into_iter()
             .filter_map(|contact_info| {
-                Some((
-                    Pubkey::from_str(&contact_info.pubkey).ok()?,
-                    contact_info.tpu?,
-                ))
+                let pubkey = Pubkey::from_str(&contact_info.pubkey).ok()?;
+                let socket = match protocol {
+                    Protocol::QUIC => contact_info.tpu_quic.or_else(|| {
+                        let mut socket = contact_info.tpu?;
+                        let port = socket.port().checked_add(QUIC_PORT_OFFSET)?;
```

Comment:
Looks like now it is the only place where this constant is used, which is good

Reply:
yeah, this is temporary as well.
it will be removed once the cluster is upgraded to the new contact-info.